api/v1alpha1/gateway_types.go

```diff
@@ -117,7 +117,7 @@ type BfdSpec struct {
 // BgpAuth defines the parameters to configure BGP authentication
 type BgpAuth struct {
 	// +kubebuilder:validation:Type=string
-	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9\.-_]+`
+	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9\.-_]+$`
```

Comment:
The pattern doesn't accept: "abc-def" while `-` must be accepted (https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object)
```suggestion
	// +kubebuilder:validation:Pattern=``^[-._a-zA-Z0-9]+$``
```
I got this pattern when I tried to put the wrong key in a secret:
```
The Secret "test" is invalid: data[abc+def]: Invalid value: "abc+def": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')
```

Reply:
Right, the dash was not escaped thus got interpreted as range.